src/shared-components/segmentedControl/index.tsx

```diff
@@ -58,6 +54,10 @@ export const SegmentedControl: React.FC<SegmentedControlProps> = ({
     }
   };
 
+  if (itemsCount === 0) {
```

Comment:
All hooks must run in the same order, and in their entirety. Early returns violate this rule.